### ISSUE TITLE
no-inferred-empty-object: doesn't detect JSDoc function type

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#235]
```js
/** @type {<T>(param: T) => T} */
export var functionTyped;
```
